src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,465 @@
+        enddo
+    end function unif_dist_rvs_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    function unif_dist_rvs_array_${t1[0]}$${k1}$(loc, scale, array_size)        &
+             result(res)
+        ${t1}$, intent(in) :: loc, scale
+        ${t1}$, allocatable :: res(:)
+        real(${k1}$) :: r1, r2, tr, ti
+        integer, intent(in) :: array_size
+        integer(int64) :: tmp
+        integer :: i
+
+
+        if(scale == (0.0_${k1}$, 0.0_${k1}$)) call error_stop("Error(unif_dist_"&
+           //"rvs_array): Uniform distribution scale parameter must be non-zero")
+        allocate(res(array_size))
+        do i = 1, array_size
+            tmp = shiftr(dist_rand(INT_ONE), 11)
+            r1 = real(tmp * MESENNE_NUMBER, kind = ${k1}$)
+            if(real(scale) == 0.0_${k1}$) then
+                tr = real(loc)
+                ti = aimag(loc) + aimag(scale) * r1
+            elseif(aimag(scale) == 0.0_${k1}$) then
+                tr = real(loc) + real(scale) * r1
+                ti = aimag(loc)
+            else
+                tmp = shiftr(dist_rand(INT_ONE), 11)
+                r2 = real(tmp * MESENNE_NUMBER, kind = ${k1}$)
+                tr = real(loc) + real(scale) * r1
+                ti = aimag(loc) + aimag(scale) * r2
+            endif
+            res(i) = cmplx(tr, ti, kind=${k1}$)
+        end do
+    end function unif_dist_rvs_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    elemental function unif_dist_pdf_${t1[0]}$${k1}$(x, loc, scale) result(res)
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+
+        if(scale == 0) then
+            res = 0.0
+        elseif(x < loc .or. x > (loc + scale)) then
+            res = 0.0
+        else
+            res = 1. / (scale + 1_${k1}$)
+        end if
+    end function unif_dist_pdf_${t1[0]}$${k1}$
+
+    #:endfor
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    elemental function unif_dist_pdf_${t1[0]}$${k1}$(x, loc, scale) result(res)
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+
+        if(scale == 0.0_${k1}$) then
+            res = 0.0
+        elseif(x <= loc .or. x >= (loc + scale)) then
+            res = 0.0
+        else
+            res = 1.0 / scale
+        end if
+    end function unif_dist_pdf_${t1[0]}$${k1}$
+
+    #:endfor
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    elemental function unif_dist_pdf_${t1[0]}$${k1}$(x, loc, scale) result(res)
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+        real(${k1}$) :: tr, ti
+
+        tr = real(loc) + real(scale); ti = aimag(loc) + aimag(scale)
+        if(scale == (0.0_${k1}$,0.0_${k1}$)) then
+            res = 0.0
+        elseif((real(x) >= real(loc) .and. real(x) <= tr) .and.                 &
+               (aimag(x) >= aimag(loc) .and. aimag(x) <= ti)) then
+            res = 1.0 / (real(scale) * aimag(scale))
+        else
+            res = 0.0
```

Comment:
Should it not be 1., as for `int` and `real` related functions.

Reply:
No. this is the `_pdf` probability function, not cumulative distribution function. Actually it is the same as `int` and `real` related  definitions just above it.

Reply:
Ok. Thank you for the explanation.